pkg/gce-pd-csi-driver/controller.go

```diff
@@ -716,6 +760,80 @@ func (gceCS *GCEControllerServer) createSingleDisk(ctx context.Context, req *csi
 }
 
+func (gceCS *GCEControllerServer) ControllerModifyVolume(ctx context.Context, req *csi.ControllerModifyVolumeRequest) (*csi.ControllerModifyVolumeResponse, error) {
+	var err error
+
+	volumeID := req.GetVolumeId()
+	klog.V(4).Infof("Modifying Volume ID: %s", volumeID)
+
+	if volumeID == "" {
+		return nil, status.Error(codes.InvalidArgument, "volume ID must be provided")
+	}
+
+	diskType := metrics.DefaultDiskTypeForMetric
+	enableConfidentialCompute := metrics.DefaultEnableConfidentialCompute
+	enableStoragePools := metrics.DefaultEnableStoragePools
+
+	defer func() {
+		gceCS.Metrics.RecordOperationErrorMetrics("ModifyVolume", err, diskType, enableConfidentialCompute, enableStoragePools)
+	}()
+
+	project, volKey, err := common.VolumeIDToKey(volumeID)
+
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "volume ID is invalid: %v", err.Error())
+	}
+
+	volumeModifyParams, err := common.ExtractModifyVolumeParameters(req.GetMutableParameters())
+	if err != nil {
+		klog.Errorf("Failed to extract parameters for volume %s: %v", volumeID, err)
+		return nil, status.Errorf(codes.InvalidArgument, "Invalid parameters: %v", err)
+	}
+	klog.V(4).Infof("Modify Volume Parameters for %s: %v", volumeID, volumeModifyParams)
+
+	existingDisk, err := gceCS.CloudProvider.GetDisk(ctx, project, volKey, gce.GCEAPIVersionBeta)
+
+	if err != nil {
+		return nil, status.Errorf(codes.NotFound, "failed to get volume: %v", err)
+	}
+
+	if existingDisk == nil || existingDisk.GetSelfLink() == "" {
+
+		return nil, status.Errorf(codes.Internal, "failed to get volume : %s", volumeID)
+	}
+	diskType = existingDisk.GetPDType()
+
+	// Check if the disk is a hyperdisk (supports dynamic provisioning)
+	if _, valid := diskSupportsIopsChange[diskType]; !valid {
+		return nil, status.Errorf(codes.InvalidArgument, "Failed to modify volume: modifications not supported for disk type %s", diskType)
+	}
+	supportsIopsChange := diskSupportsIopsChange[diskType]
+	supportsThroughputChange := diskSupportsThroughputChange[diskType]
+	if !supportsIopsChange && !supportsThroughputChange {
+		return nil, status.Errorf(codes.InvalidArgument, "Failed to modify volume: modifications not supported for disk type %s", diskType)
+	}
+	if !supportsIopsChange && volumeModifyParams.IOPS != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "Cannot specify IOPS for disk type %s", diskType)
+	}
+	if !supportsThroughputChange && volumeModifyParams.Throughput != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "Cannot specify throughput for disk type %s", diskType)
+	}
+
+	if existingDisk.GetEnableStoragePools() {
+		enableStoragePools = "enabled"
+	} else {
+		enableStoragePools = "disabled"
+	}
+
+	err = gceCS.CloudProvider.UpdateDisk(ctx, project, volKey, existingDisk, volumeModifyParams)
+	if err != nil {
+		klog.Errorf("Failed to modify volume %s: %v", volumeID, err)
+		return nil, status.Errorf(codes.Internal, "Failed to modify volume %s: %v", volumeID, err)
```

Comment:
Same thing here, please wrap the error. It might not be an internal error (it could be an InvalidArgument, maybe the provided iops/throughput were set too low/high).

Reply:
Done